src/role_sourcer.js

```diff
@@ -19,9 +19,14 @@ roles.sourcer.killPrevious = true;
 // TODO should be true, but flee must be fixed before 2016-10-13
 roles.sourcer.flee = false;
 
-roles.sourcer.getPartConfig = function(room, energy, heal) {
-  var parts = [MOVE, CARRY, WORK, WORK, WORK, WORK, MOVE, MOVE, WORK, HEAL, MOVE, HEAL, MOVE, WORK, WORK, WORK, WORK, WORK, MOVE, MOVE, MOVE, MOVE, MOVE];
-  return room.getPartConfig(energy, parts);
+roles.sourcer.getPartConfig = function(room, creep) {
+  let datas = {layout: [MOVE, CARRY, WORK, WORK, WORK, WORK, MOVE, MOVE, WORK],
+    minEnergyStored: 200};
+  if (creep.heal) {
+    datas.layout = datas.layout.concat([HEAL, MOVE, HEAL, MOVE, WORK, WORK, WORK,
+      WORK, WORK, MOVE, MOVE, MOVE, MOVE, MOVE]);
+  }
```

Comment:
i was more thinking about improve the number of MOVE watching fatigue needed for travel. because it's nor really usefull to have a 15*MOVE sourcer in base room.